### ISSUE TITLE
HTTP content compare - multiregion compare probe

### ISSUE THREAD

**Issue by pdunk** [prometheus/blackbox_exporter#273]
**Feature Request/Proposal**
Hi,
we started building a new infrastructure with K8s deployed in multiple AWs regions and running these in active-active mode. Prometehus/grafana is core for monitoring the environments.
We want to monitor now also that the results of our API in multiple regions are identical (similar enough) and add a similarity or compare metric, e.g based on levenstein distance.

**Curious if such feature would be interesting to others?**

I actually implemented this now myself as an extension in blackbox exporter, but based on feedback would change the code before making an official proposal or PR

Different options how to add such feature would be
A) Add a 'reference_url' in config/query parameter which is loaded in parallel and the content of target and reference is compared and an additional metric added to the current http prober metrics
Drawback: would need to add a 'reference' string parameter in the prober function definition

B) Add a new url function such as /compare in parallel to /probe, this would allow separating the code more, but likely do more code duplication and would require operationally to probe all URL's twice

C) Add a new probe type, such as http_compare, which still requires expanding the prober function reference with a 'reference' parameter, but otherwise, it is just another instance of prober and keeps code nicely separate --> that's how I implemented it right now, still some code duplication to the http part though

D) You guys think this makes no sense at all in blackbox_exporter and you hate the idea, which would be too bad :-(

**Comment by brian-brazil** [comment]
This sounds fairly niche, and probably out of scope for the blackbox exporter. I think this is a case where a separate exporter would be best, especially if you want things like levenstein distance.

**Comment by pdunk** [comment]
I am happy to contribute, not sure what your procedure for this is though. I did code it already and use the smetrics library for the levenstein part.
I heard the comment before that this seems niche, maybe it is more common in QA/QC tools, but I am wondering how come or if others dont have a similar problem on a operational side.

**Comment by brian-brazil** [comment]
The purpose of the blackbox exporter is to know if something is very roughly speaking working. This is going quite a bit beyond that, and is kinda like reading part of the config file from a remote server which would be against how we generally do things.
This really sounds like something best done in a separate exporter.
